### PR TITLE
Allow pot task to take a filename

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

File: lib/tasks/gettext.rake [hunk 1]
```diff
@@ -18,11 +18,13 @@ namespace :gettext do
   end
 
   desc 'Generate POT file'
-  task :pot do
+  task :pot, [:filename] do |_, args|
     begin
-      result = GettextSetup::Pot.generate_new_pot
+      filename = args.filename || ENV['filename']
```
Comment (smcclellan) [inline lib/tasks/gettext.rake]: Do we want to read this generic environment variable? I think standardizing on just one way to call the task is fine.

File: lib/gettext-setup/pot.rb [hunk 2]
```diff
@@ -50,7 +50,7 @@ def self.string_changes?(old_pot, new_pot)
       return true
     end
 
-    def self.generate_new_pot(locales_path = GettextSetup.locales_path, path = nil)
+    def self.generate_new_pot(path = nil, locales_path = GettextSetup.locales_path)
```
Comment (smcclellan) [inline lib/gettext-setup/pot.rb]: Why are we changing the order of these? They're currently consistent with `update_pot`.

Comment (smcclellan) [reply lib/gettext-setup/pot.rb]: I'm fine with switching (omitting the second argument in later calls makes sense), I think it's worth keeping these in sync though.

File: lib/tasks/gettext.rake (hunk 1 above)
Comment (eputnam) [reply lib/tasks/gettext.rake]: @tphoney , I wasn't clear on what we needed `ENV['filename']` for or not, so I just left it. Perhaps we can pitch it?

File: lib/tasks/gettext.rake [hunk 3]
```diff
@@ -18,11 +18,13 @@ namespace :gettext do
   end
 
   desc 'Generate POT file'
-  task :pot do
+  task :pot, [:filename] do |_, args|
     begin
-      result = GettextSetup::Pot.generate_new_pot
+      filename = args.filename
+      result = GettextSetup::Pot.generate_new_pot (path: filename)
```
Comment (tphoney) [inline lib/tasks/gettext.rake]: The reason i changed the ordering, was because named arguments was failing. You can now see: @smcclellan @eputnam

**Comment by smcclellan** [comment]
I squashed the three commits as https://github.com/puppetlabs/gettext-setup-gem/commit/d853865b089b0c0090727707e316234d4b072054.

**Pull request merged**